lib/msf/base/simple/exploit.rb

```diff
@@ -69,7 +69,7 @@ def self.exploit_simple(oexploit, opts, &block)
 
       # Make sure parameters are valid.
       if (opts['Payload'] == nil)
-        raise MissingPayloadError.new, caller
+        raise MissingPayloadError, 'A payload has not been selected.', caller
```

Comment:
When `caller` is provided, it must be the third argument. Not the second. In the second position it's treated as the error message.